### ISSUE TITLE
Ambiguous steps

### ISSUE THREAD

**Issue by lloydwatkin** [acuminous/yadda#107]
Hi,

Having an issue with step definitions which I'll give an example for and then explain some extra stuff. Sorry if these examples are bad, but they are just to provide a guide.

If I'm writing a scenario sometimes I may want to define it like this:

```
Given I click the 'Logout' link
When I visit the home page
Then I am redirected to the login page
```

Then consider:

```
Given I am on the home page
When I click the 'Logout' link
Then I am redirected to the login page
```

Both similar tests (but both could be different in some ways). However the important point is that it is equally valid to use the step **Given/When I click the 'Logout' link**.  

At present I have to define two steps, one beginning with `when()` and the other with `given()`. Immediately we have an issue with duplicated steps.

Additionally if I were to then write something like this:

```
Given I am on the home page
When I click the 'remember me' button
And I click the 'Logout' link
Then I am redirected to the login page
```

I now have the issue that my scenario won't run as this **And** step is seen as ambiguous by the step definitions (I load all of my step definition files for all tests).

The current workaround is just not to use **And** and use a **Given** or **When** as appropriate, however this doesn't seem ideal.

I can think of two solutions to this issue:
- Update the feature file parser to automatically convert an **And** to whatever came previous to it (so i the above case a **When**). However this raises issues as it doesn't seem to be invalid to start a scenario with **And** in **Yadda** (which makes sense if you have a **Background**
- Ignore the **Given**, **When**, **Then** in the step definition resolver. This solves the above issue and also means that I don't have to replicate steps.  I've talked to a friend who works in automated testing and it seems that this is what **watir-webdriver** does in **Ruby**

Interested to see what you think and potentially look at getting one of the above implemented if possible?

Thanks for your great work!

**Comment by cressie176** [comment]
what about if I add an extra library alias for Given / When and Then, e.g.

```js
library.given_when_then(/some step/)
```
If this is too wordy (I think it is) I could make it one of
```js
library.gwt(/some step)
library.any(/some step/)
library.step(/some step/)
```

**Comment by lolmaus** [comment]
> `Given I click the 'Logout' link`

This is wrong, IMO. `Given` is for predefined conditions, such as DB state. User interactions should be prefixed with `When`.

PS Good to close?
